docs/content/2.deploy/providers/firebase.md

```diff
@@ -82,11 +104,86 @@ NITRO_PRESET=firebase yarn build
 firebase emulators:start
 ```
 
-## Deploy to Firebase Hosting via CLI
+## Build and Deploy
 
-Deploying to Firebase Hosting is a simple matter of just running the `firebase deploy` command.
+Deploy to Firebase Hosting by running a nitro build and then running the `firebase deploy` command.
 
 ```bash
 NITRO_PRESET=firebase yarn build
+```
+
+```bash
+npx firebase-tools deploy
+```
+
+If you chose the alternative setup using the Firebase CLI, you can also run:
+
+```bash
 firebase deploy
 ```
+
+## Using 2nd Generation Firebase Functions
+
+- [Comparison between 1st and 2nd generation functions](https://firebase.google.com/docs/functions/version-comparison)
+
+To switch to the more recent and, recommended generation of firebase functions, set the `FIREBASE_FUNCTIONS_GEN` to `2` when building:
+
+```bash
+FIREBASE_FUNCTIONS_GEN=2 NITRO_PRESET=firebase yarn build
+```
+
+If you already have a deployed version of your website and want to upgrade to 2nd gen, [see the Migration process on Firebase docs](https://firebase.google.com/docs/functions/2nd-gen-upgrade). Namely, the CLI will ask you to delete your existing functions before deploying the new ones.
+
+## Options
+
+You can set options for the firebase functions in your `nitro.config.ts` file:
+
+```ts
+import { defineNitroConfig } from 'nitropack/config'
+
+export default defineNitroConfig({
+  // Nitro options...
+  firebase: {
+    gen2: {
```

Comment:
Just double checking, is `genX` a known term for firebase team/docs that we use here? (i love it btw!)

Reply:
We can also make types smart by supporting `generation: 1|2` to switch types for rest of the keys. This way we don't need to make env required.

Reply:
> Just double checking, is genX a known term for firebase team/docs that we use here? (i love it btw!)

They actually use _nth gen_ like 1st gen but I thought that having the number instead of letters was easier to spot 😅 . Docs with terms: https://firebase.google.com/docs/functions/version-comparison

> We can also make types smart by supporting generation: 1|2 to switch types for rest of the keys. This way we don't need to make env required.

That's pretty cool but I don't know how to add it 😆

Reply:
Good to know abouth nth! Example for selective type: https://github.com/unjs/unbuild/blob/main/src/types.ts#L41